### PR TITLE
 [FLINK-11608] [docs] Translate the "Local Setup Tutorial" page into Chinese

### PR DESCRIPTION
**Pull request by shengjk**
## What is the purpose of the change
Translate the "Local Setup Tutorial" page into Chinese

## Brief change log
  -  Add docs/tutorials/local_setup.zh.md

### CODE REVIEW

**Comment by dawidwys** [comment]
@flinkbot approve description
@flinkbot approve consensus
@flinkbot attention @wuchong

**Comment by wuchong** [comment]
Thanks for the reminder @dawidwys . But this pull request should be postponed until #7754 is merged.

Thanks for the work @shengjk . Please rebase your branch on top of the latest master and avoid "merge" commits.

**Comment by dawidwys** [comment]
Sure, I didn't mean to remind or rush you @wuchong ;). Just tagged you according to our review process. Take as long time as you need.

**Comment by shengjk** [comment]
ok @wuchong ,thank you

File: docs/tutorials/local_setup.zh.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+通过几个简单的步骤来启动并运行 Flink 程序。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
只需几个简单的步骤即可启动并运行 Flink 示例程序
```

File: docs/tutorials/local_setup.zh.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+## 安装：下载并启动 Flink 
+
+Flink 可以在 __Linux, Mac OS X, and Windows__ 环境中运行。要想运行 Flink 仅仅需要安装 __Java 8.x__ 。 Windows 用户， 请查看 [Flink on Windows]({{ site.baseurl }}/tutorials/flink_on_windows.html)  上面描述了如何在 windows 上以本地模式运行 Flink。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
Flink 可以在 __Linux、Mac OS X,，和 Windows__ 环境中运行。为了能够运行 Flink，唯一的要求是安装 __Java 8.x__。 Windows 用户， 请查看 [Flink on Windows]({{ site.baseurl }}/tutorials/flink_on_windows.html) 指南，该指南介绍了如何在 windows 上以本地设置运行 Flink。
```

File: docs/tutorials/local_setup.zh.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+你可以用下面的命令来检查一下是否正确的安装了 Java 程序：
+
+{% highlight bash %}
+java -version
+{% endhighlight %}
+
+如果你已经安装了 Java 8，它的输出应该是这个样子的：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
如果你已经安装了 Java 8，应该输出如下内容：
```

File: docs/tutorials/local_setup.zh.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+{% highlight bash %}
+java version "1.8.0_111"
+Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
+Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)
+{% endhighlight %}
+
+{% if site.is_stable %}
+<div class="codetabs" markdown="1">
+<div data-lang="Download and Unpack" markdown="1">
+1. 从 [downloads page](http://flink.apache.org/downloads.html)下载二进制文件。你可以选择任何你喜欢的 Hadoop/Scala 组合。如果你仅仅想在本地使用，那么任何的 Hadoop 版本都可以。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
1. 从 [下载页](http://flink.apache.org/downloads.html)下载二进制文件。你可以选择任何喜欢的 Scala 版本。针对某些特性，你可能还需要下载预捆绑的 Hadoop jar 包并将它们放入 `/lib`  目录。
```

File: docs/tutorials/local_setup.zh.md [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+2. 去下载后的目录。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
2. 进入下载的目录。
```

File: docs/tutorials/local_setup.zh.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+
+{% highlight bash %}
+$ cd ~/Downloads        # Go to download directory
+​$ tar xzf flink-*.tgz   # Unpack the downloaded archive
+$ cd flink-{{site.version}}
+{% endhighlight %}
+</div>
+
+<div data-lang="MacOS X" markdown="1">
+对于 MacOS X 用户, Flink 可以通过 [Homebrew](https://brew.sh/)进行安装。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
对于 MacOS X 用户，Flink 可以通过 [Homebrew](https://brew.sh/)进行安装。
```

File: docs/tutorials/local_setup.zh.md [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+​$ flink --version
+Version: 1.2.0, Commit ID: 1c659cf
+{% endhighlight %}
+</div>
+
+</div>
+
+{% else %}
+### 下载和编译
+从我们的 [repositories](http://flink.apache.org/community.html#source-code) 中下载源码，比如：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
从我们的某个 [代码仓库](http://flink.apache.org/community.html#source-code) 中克隆源码，例如：
```

File: docs/tutorials/local_setup.zh.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+{% endhighlight %}
+{% endif %}
+
+### 启动 Flink 本地集群
+
+{% highlight bash %}
+$ ./bin/start-cluster.sh  # Start Flink
+{% endhighlight %}
+
+通过  [http://localhost:8081](http://localhost:8081) 检查 __Dispatcher's web frontend__ 以确保所有的程序已经启动并正在运行。Web 界面上会显示一个可用的 TaskManager 实例。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
检查  [http://localhost:8081](http://localhost:8081) 的 __web 调度界面__ 以确保一切正常运行。Web 界面上会仅显示一个可用的 TaskManager 实例。
```

File: docs/tutorials/local_setup.zh.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+
+<a href="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-1.png" ><img class="img-responsive" src="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-1.png" alt="Dispatcher: Overview"/></a>
+
+你也可以检查 log 文件来确认程序是否正在运行：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
还可以通过检查 `logs` 目录中的日志文件来验证系统是否正在运行：
```

File: docs/tutorials/local_setup.zh.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+INFO ... - ResourceManager akka.tcp://flink@localhost:6123/user/resourcemanager was granted leadership ...
+INFO ... - Starting the SlotManager.
+INFO ... - Dispatcher akka.tcp://flink@localhost:6123/user/dispatcher was granted leadership ...
+INFO ... - Recovering all persisted jobs.
+INFO ... - Registering TaskManager ... under ... at the SlotManager.
+{% endhighlight %}
+
+## 阅读代码
+
+你可以在 Github 上看到分别用  [scala](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/socket/SocketWindowWordCount.scala) and [java](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/socket/SocketWindowWordCount.java) 书写的 SocketWindowWordCount 的完整样例
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
你可以在 Github 上看到分别用  [scala](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/socket/SocketWindowWordCount.scala) 和[java](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/socket/SocketWindowWordCount.java) 书写的 SocketWindowWordCount 示例的完整源码。
```

File: docs/tutorials/local_setup.zh.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+        }
+    }
+}
+{% endhighlight %}
+</div>
+</div>
+
+## 运行样例
+
+现在，我们去运行 Flink 程序。 只要有单词输入，它就从一个套接字中读取文本，每5秒打印一次前5秒中每个不同单词的出现次数，也就是一个 tumbling window 的处理时间
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
现在，我们将运行此 Flink 程序。 只要有单词输入，它就从一个套接字中读取文本，每 5 秒打印一次前 5 秒中每个不同单词的出现次数，5秒钟也就是处理时间的翻滚窗口。
```

File: docs/tutorials/local_setup.zh.md [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+* 提交 Flink 程序：
+
+{% highlight bash %}
+$ ./bin/flink run examples/streaming/SocketWindowWordCount.jar --port 9000
+Starting execution of program
+
+{% endhighlight %}
+
+  这个程序会连接 socket  服务并等待输入。你可以通过 web interface 来验证 job 是否如期运行:
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
程序连接到套接字并等待输入。你可以通过 web 界面来验证 job 是否如期运行：
```

File: docs/tutorials/local_setup.zh.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+  <div class="row">
+    <div class="col-sm-6">
+      <a href="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-2.png" ><img class="img-responsive" src="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-2.png" alt="Dispatcher: Overview (cont'd)"/></a>
+    </div>
+    <div class="col-sm-6">
+      <a href="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-3.png" ><img class="img-responsive" src="{{ site.baseurl }}/page/img/quickstart-setup/jobmanager-3.png" alt="Dispatcher: Running Jobs"/></a>
+    </div>
+  </div>
+
+* 单词以5秒的时间窗口 (processing time, tumbling windows) 来计数，并打印到 stdout。监视 TaskManager的输出文件和 nc 的输入（输入的内容被逐行发送到 flink ） <RETURN>):
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
* 单词以5秒的时间窗口（处理时间，翻滚窗口）来计数，并打印到 `stdout`。监视 TaskManager 的输出文件并在 `nc` 中输入（输入的内容被逐行发送到 Flink）：
```

File: docs/tutorials/local_setup.zh.md [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+
+{% highlight bash %}
+$ nc -l 9000
+lorem ipsum
+ipsum ipsum ipsum
+bye
+{% endhighlight %}
+
+  当有单词输入的时候， `.out` 文件将在每个时间窗口结束时打印结果, 比如：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
随着单词的输入， `.out` 文件会在每个时间窗口的末尾打印计数，比如：
```

File: docs/tutorials/local_setup.zh.md [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+
+{% highlight bash %}
+$ tail -f log/flink-*-taskexecutor-*.out
+lorem : 1
+bye : 1
+ipsum : 4
+{% endhighlight %}
+
+  停止 Flink：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
完成输入后停止 Flink：
```

File: docs/tutorials/local_setup.zh.md [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+
+{% highlight bash %}
+$ ./bin/stop-cluster.sh
+{% endhighlight %}
+
+## 下一步
+
+查看更多的 [examples]({{ site.baseurl }}/examples) 以便更好的理解 Flink's APIs。当你做完这些的时候, 就可以继续向前，去读 [streaming 指南]({{ site.baseurl }}/dev/datastream_api.html).
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
查看更多[示例]({{ site.baseurl }}/examples) 以更好地了解 Flink 的编程 API。完成后，请继续阅读[流处理指南]({{ site.baseurl }}/zh/dev/datastream_api.html)。
```

File: docs/tutorials/local_setup.zh.md [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+只需要几个简单的步骤即可启动并运行 Flink 示例程序。
+
+## 安装：下载并启动 Flink 
+
+Flink 可以在 __Linux、 Mac OS X、 和 Windows__ 环境中运行。为了能够运行 Flink 唯一的要求是安装 __Java 8.x__ 。 Windows 用户， 请查阅[在  Windows 上运行 Flink]({{ site.baseurl }}/tutorials/flink_on_windows.html)  上面描述了如何在 windows 上以本地模式运行 Flink。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
Flink 可以在 __Linux、Mac OS X、和 Windows__ 环境中运行。为了能够运行 Flink 唯一的要求是安装 __Java 8.x__。Windows 用户，请查阅[在  Windows 上运行 Flink]({{ site.baseurl }}/tutorials/flink_on_windows.html)，上面描述了如何在 windows 上以本地模式运行 Flink。
```

File: docs/tutorials/local_setup.zh.md [hunk 18]
```diff
@@ -0,0 +1,288 @@
...
+{% highlight bash %}
+java version "1.8.0_111"
+Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
+Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)
+{% endhighlight %}
+
+{% if site.is_stable %}
+<div class="codetabs" markdown="1">
+<div data-lang="Download and Unpack" markdown="1">
+1. 从[下载页](http://flink.apache.org/downloads.html)下载二进制文件。你可以选择任何喜欢的 Scala 版本。针对某些特性，你可能还需要下载预捆绑的 Hadoop jar 包并将它们放入 `/lib`  目录。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
1. 从[下载页](http://flink.apache.org/downloads.html)下载二进制文件。你可以选择任何喜欢的 Scala 版本。针对某些特性，你可能还需要下载预捆绑的 Hadoop jar 包并将它们放入 `/lib` 目录。
```

File: docs/tutorials/local_setup.zh.md [hunk 19]
```diff
@@ -0,0 +1,288 @@
...
+
+{% highlight bash %}
+$ cd ~/Downloads        # Go to download directory
+$ tar xzf flink-*.tgz   # Unpack the downloaded archive
+$ cd flink-{{site.version}}
+{% endhighlight %}
+</div>
+
+<div data-lang="MacOS X" markdown="1">
+对于 MacOS X 用户，Flink 可以通过[Homebrew](https://brew.sh/)进行安装。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
对于 MacOS X 用户，Flink 可以通过 [Homebrew](https://brew.sh/) 进行安装。
```

Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
对于 MacOS X 用户，Flink 可以通过 [Homebrew](https://brew.sh/) 进行安装。
```

File: docs/tutorials/local_setup.zh.md [hunk 20]
```diff
@@ -0,0 +1,288 @@
...
+{% endhighlight %}
+{% endif %}
+
+### 启动 Flink 本地集群
+
+{% highlight bash %}
+$ ./bin/start-cluster.sh  # Start Flink
+{% endhighlight %}
+
+检查位于[http://localhost:8081](http://localhost:8081)的 __web 调度界面__以确保一切正常运行。Web 界面上会仅显示一个可用的 TaskManager 实例。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
检查位于 [http://localhost:8081](http://localhost:8081) 的 __web 调度界面__以确保一切正常运行。Web 界面上会仅显示一个可用的 TaskManager 实例。
```

File: docs/tutorials/local_setup.zh.md [hunk 21]
```diff
@@ -0,0 +1,288 @@
...
+
+完成输入后停止 Flink：
+
+{% highlight bash %}
+$ ./bin/stop-cluster.sh
+{% endhighlight %}
+
+## 下一步
+
+查看更多的[示例]({{ site.baseurl }}/examples)以便更好的理解 Flink 的编程的 APIs。完成后, 请继续阅读[流处理指南]({{ site.baseurl }}/zh/dev/datastream_api.html)。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
查看更多的[示例]({{ site.baseurl }}/examples)以便更好的理解 Flink 的编程 API。完成后, 请继续阅读[流处理指南]({{ site.baseurl }}/zh/dev/datastream_api.html)。
```

Comment (YngwieWang) [reply docs/tutorials/local_setup.zh.md]: 建议“的”能省略尽量省略，英文用单数形式。

File: docs/tutorials/local_setup.zh.md [hunk 22]
```diff
@@ -0,0 +1,288 @@
...
+* 单词以5秒为时间窗口（处理时间，翻滚窗口）进行计数，并打印到 `stdout`。监视 TaskManager 的输出文件并在 `nc` 中输入一些文本（输入的内容被逐行发送到 Flink）：
+
+{% highlight bash %}
+$ nc -l 9000
+lorem ipsum
+ipsum ipsum ipsum
+bye
+{% endhighlight %}
+
+随着单词的输入，`.out`  文件会在每个时间窗口的末尾打印计数，比如：
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
随着单词的输入，`.out` 文件会在每个时间窗口的末尾打印计数，比如：
```

**Comment by shengjk** [comment]
@YngwieWang ok, thank you

File: docs/tutorials/local_setup.zh.md [hunk 23]
```diff
@@ -0,0 +1,288 @@
...
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+只需要几个简单的步骤即可启动并运行 Flink 示例程序。
+
+## 安装：下载并启动 Flink 
+
+Flink 可以在 __Linux、Mac OS X、和 Windows__ 环境中运行。为了能够运行 Flink 唯一要求是安装 __Java 8.x__ 。 Windows 用户，请查阅[在  Windows 上运行 Flink]({{ site.baseurl }}/tutorials/flink_on_windows.html)上面描述了如何在 windows 上以本地模式运行 Flink。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
Flink 可以在 __Linux、Mac OS X 和 Windows__ 环境中运行。运行 Flink 的唯一要求是安装 __Java 8.x__。Windows 用户请查阅[在  Windows 上运行 Flink]({{ site.baseurl }}/zh/tutorials/flink_on_windows.html)，上面描述了如何在 windows 上以本地模式运行 Flink。
```

File: docs/tutorials/local_setup.zh.md [hunk 24]
```diff
@@ -0,0 +1,288 @@
...
+INFO ... - ResourceManager akka.tcp://flink@localhost:6123/user/resourcemanager was granted leadership ...
+INFO ... - Starting the SlotManager.
+INFO ... - Dispatcher akka.tcp://flink@localhost:6123/user/dispatcher was granted leadership ...
+INFO ... - Recovering all persisted jobs.
+INFO ... - Registering TaskManager ... under ... at the SlotManager.
+{% endhighlight %}
+
+## 阅读代码
+
+你可以在 Github 上看到分别用[scala](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/socket/SocketWindowWordCount.scala)和[java](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/socket/SocketWindowWordCount.java)编写的 SocketWindowWordCount 的完整代码。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
你可以在 Github 上看到分别用 [scala](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/scala/org/apache/flink/streaming/scala/examples/socket/SocketWindowWordCount.scala) 和 [java](https://github.com/apache/flink/blob/master/flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/socket/SocketWindowWordCount.java) 编写的 SocketWindowWordCount 示例程序的完整代码。
```

File: docs/tutorials/local_setup.zh.md [hunk 25]
```diff
@@ -0,0 +1,288 @@
...
+
+完成输入后停止 Flink：
+
+{% highlight bash %}
+$ ./bin/stop-cluster.sh
+{% endhighlight %}
+
+## 下一步
+
+查看更多的[示例]({{ site.baseurl }}/examples)以便更好的理解 Flink 的 API。完成后, 请继续阅读[流处理指南]({{ site.baseurl }}/zh/dev/datastream_api.html)。
```
Comment (YngwieWang) [inline docs/tutorials/local_setup.zh.md]:
```suggestion
查看更多的[示例]({{ site.baseurl }}/examples)以便更好地理解 Flink 的 API。完成后, 请继续阅读[流处理指南]({{ site.baseurl }}/zh/dev/datastream_api.html)。
```

File: docs/tutorials/local_setup.zh.md (hunk 24 above)
Comment (YngwieWang) [reply docs/tutorials/local_setup.zh.md]: image
运行一下 `build_docs.sh` 看看效果，英文中括号后面跟小括号是超链接，如果超链接的文本内容是英文，要和前后的中文之间留有空格。

**Comment by shengjk** [comment]
@YngwieWang Thanks for your comment. I hava run build_docs.sh and checked it. please help me to check it, thanks again!

**Comment by shengjk** [comment]
requested a review from @wuchong

**Comment by klion26** [comment]
@shengjk thanks for your contribution. could you please get rid of `git merge` in the history, maybe you could use `git rebase` instead. after that, I can help review.

**Comment by shengjk** [comment]
> @shengjk thanks for your contribution. could you please get rid of `git merge` in the history, maybe you could use `git rebase` instead. after that, I can help review.

Thank you. maybe this PR should be close. I find 1.11 doc has a lot of changes. I will close tht PR

**Pull request closed without merging**